src/common/types/ShareClassId.sol

```diff
@@ -18,7 +18,7 @@ function raw(ShareClassId scId) pure returns (bytes16) {
 }
 
 function newShareClassId(PoolId poolId, uint32 index) pure returns (ShareClassId scId) {
-    return ShareClassId.wrap(bytes16(uint128(PoolId.unwrap(poolId)) + index));
+    return ShareClassId.wrap(bytes16(uint128(PoolId.unwrap(poolId) << 64) + index));
```

Comment:
CR: 
```suggestion
    return ShareClassId.wrap(bytes16((uint128(PoolId.unwrap(poolId)) << 64) + index));
```

Reply:
If not, we're shifting 64 bits over an `uint64`, so we're discarding the bytes. The tests are not notifying this because they assume the same